groovy/inductor_aws_hf_oob.groovy

```diff
@@ -0,0 +1,162 @@
+env.instance_name = 'spr-hf-oob'
+env.target = new Date().format('yyyy_MM_dd')
+env.LOG_DIR = 'hf_oob_log'
+
+env.terminate_instance = 'False'
+if ('terminate_instance' in params) {
+    echo "terminate_instance in params"
+    if (params.terminate_instance != '') {
+        env.terminate_instance = params.terminate_instance
+    }
+}
+echo "terminate_instance: $terminate_instance"
+
+def getGroovyEnv() {
+    def env_groovy = ''
+    env.getEnvironment().each {
+        name, value ->
+            if(value){
+                env_groovy = env_groovy + "${name}=${value.trim().replaceAll("\n", "")}\n"
+            }else{
+                env_groovy = env_groovy + "${name}=NULL\n"
+            }
+    }
+    return env_groovy
+}
+
+node(NODE_LABEL){
+    stage("Find or create instance"){
+        deleteDir()
+        checkout scm
+        if  ("${report_only}" == "false")
+        {
+            sh'''
+            #!/usr/bin/env bash
+            cd ${WORKSPACE}/scripts/aws/
+            while true;
+            do
+                bash find_instance.sh ${instance_name} ${instance_ids} 2>&1 | tee ${WORKSPACE}/instance_id.txt
+                ins_id=`cat ${WORKSPACE}/instance_id.txt`
+                if [ $ins_id != "waiting_instance" ]; then
+                    echo "ins_id : $ins_id"
+                    break
+                else
+                    echo "Waiting for avaliable instance, will check after 10 min..."
+                    sleep 10m
+                fi
+            done
+            '''
+        } else {
+            echo "report_only mode, will directly use the instance_id"
+            sh'''
+            #!/usr/bin/env bash
+            echo ${instance_ids} > ${WORKSPACE}/instance_id.txt
+            echo "ins_id : ${instance_ids}"
+            '''
+        }
+    }
+    stage("start instance")
+    {
+        sh '''
+        #!/usr/bin/env bash
+        ins_id=`cat ${WORKSPACE}/instance_id.txt`
+        cd $HOME && $aws ec2 start-instances --instance-ids ${ins_id} --profile pytorch && sleep 2m
+        init_ip=`$aws ec2 describe-instances --instance-ids ${ins_id} --profile pytorch --query 'Reservations[*].Instances[*].PublicDnsName' --output text`
+        echo init_ip is $init_ip
+        ssh -o StrictHostKeyChecking=no ubuntu@${init_ip} "pwd"
+        '''
+    }
+    stage("prepare scripts & benchmark") {
+        def env_groovy = getGroovyEnv()
```

Comment:
Does this method default value of parameter depend on parameter definition on Jenkins job config directly?

Reply:
Yes, this method will dump all the parameters defined in Jenkins Job and Groovy file with `env.` prefix.